Solaris-specific patch. Adds if def(SOLARIS) case.
Do not plan to send upstream.

diff --git a/xpcom/base/nsUUIDGenerator.cpp b/xpcom/base/nsUUIDGenerator.cpp
--- a/xpcom/base/nsUUIDGenerator.cpp
+++ b/xpcom/base/nsUUIDGenerator.cpp
@@ -11,17 +11,17 @@
 #include <CoreFoundation/CoreFoundation.h>
 #else
 #include <stdlib.h>
 #include "prrng.h"
 #endif
 
 #include "nsUUIDGenerator.h"
 
-#ifdef ANDROID
+#if defined(ANDROID) || defined(SOLARIS)
 extern "C" NS_EXPORT void arc4random_buf(void*, size_t);
 #endif
 
 using namespace mozilla;
 
 NS_IMPL_ISUPPORTS(nsUUIDGenerator, nsIUUIDGenerator)
 
 nsUUIDGenerator::nsUUIDGenerator()
